modules/apps/frontend-js/frontend-js-aui-web/src/main/resources/META-INF/resources/liferay/session.js

```diff
@@ -538,58 +559,76 @@ AUI.add(
 					return time;
 				},
 
-				_onHostSessionStateChange(event) {
+				_getBanner() {
 					var instance = this;
 
-					if (event.newVal === 'warned') {
-						instance._beforeHostWarned(event);
-					}
-				},
+					var banner = instance._banner;
 
-				_renderBanner(config) {
-					var instance = this;
-
-					var remainingTimeFormatted = instance._formatTime(
-						config.remainingTime
-					);
-
-					instance._updateRemainingTimeTitle(remainingTimeFormatted);
-
-					Liferay.Util.openToast(
-						Object.assign(config, {
-							message:
-								config.remainingTime >= 0
-									? Lang.sub(instance._warningText, [
-											remainingTimeFormatted,
-									  ])
-									: instance._expiredText,
-							messageType: 'html',
-							onClick(eventPayload) {
-								var event = eventPayload.event;
-								var onClose = eventPayload.onClose;
+					if (!banner) {
+						var openToast = instance.get('openToast');
+						var toastId = 'sessionToast';
 
+						var toastDefaultConfig = {
+							onClick: ({event}) => {
 								if (
 									event.target.classList.contains(
 										'alert-link'
 									)
 								) {
-									event.preventDefault();
-
 									instance._host.extend();
-
-									onClose();
-
-									instance._alertClosed = true;
 								}
 							},
-							title: Liferay.Language.get('warning'),
+							onClose: ({event}) => {
+								event.preventDefault();
+
+								A.one(`#${toastId}`).toggleClass('hide', true);
+							},
+							renderData: {
+								componentId: toastId,
+							},
 							toastProps: {
-								autoClose: config.remainingTime,
+								autoClose: false,
+								id: toastId,
 								role: 'alert',
 							},
+						};
+
+						openToast({
+							message: instance._warningText,
 							type: 'warning',
-						})
-					);
+							...toastDefaultConfig,
+						});
+
+						var toastComponent = Liferay.component(toastId);
+
+						banner = {
+							one: A.one(`#${toastId}`).one,
+							setAttrs: (attrs) => {
+								toastComponent.destroy();
+
+								openToast({
+									...attrs,
+									...toastDefaultConfig,
+								});
+							},
+							show: () => {
```

Comment:
Two more places that would look slightly neater/briefer with object concise methods.

Reply:
It's going to look beautiful!